### ISSUE TITLE
Trace Call errors much more frequent

### ISSUE THREAD

**Issue by 0xDmtri** [paradigmxyz/reth#4832]
### Describe the bug

Hi Dear Sers!

I know it may sound like I am an idiot but after updating to Alpha 9 (and 10) I noticed that usual `client.trace_call()` started to return error: `code: -32003, message: "nonce too high", data: None` much much more frequent. I am talking about pending txs, and I do know that in some cases its perfectly normal.

I noticed that right after I updated I started getting these errors incredibly frequent, so I downgraded back to Alpha 8 and things started to look as it used to be. I still do get "nonce too high" but  so much less frequent. I mean by magnitude of 10 times at least.

I know I probably described it terribly as I don't really understand why would I see such a dramatic change in `trace_call` behavior but I definitely see it.

The only way I see to compare is to run two clients (Alpha 8 and Alpha 10) simultaneously side by side.

Attaching a small code snippet below.

Please let me know how I can help/what I can do in order to better show/understand the problem.

<img width="779" alt="Screenshot 2023-09-28 at 01 11 03" src="https://github.com/paradigmxyz/reth/assets/33504992/1d243bca-c4c3-48c1-9273-536074f9c6cf">

### Steps to reproduce

```Rust
use anyhow::Result;
use ethers::prelude::*;

#[tokio::main]
async fn main() -> Result<()> {
    let client = Provider::connect_ipc("/tmp/reth.ipc").await?;

    let mut stream = client.subscribe_full_pending_txs().await?;

    while let Some(tx) = stream.next().await {
        match client
            .trace_call(&tx, vec![TraceType::StateDiff], None)
            .await
        {
            Ok(_) => (),
            Err(err) => {
                println!("Error: {:?}", err);
                println!("For tx hash: {:?}", &tx.hash);
                println!("None from the tx: {:?}", &tx.nonce);

                let current_nonce = client
                    .get_transaction_count(NameOrAddress::Address(tx.from), None)
                    .await?;

                println!("Current nonce: {:?}", current_nonce);
            }
        }
    }

    Ok(())
}
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

Version: 0.1.0-alpha.10, Commit SHA: 1b16d804, Build Features: default, jemalloc.

### What database version are you on?

Current database version: 1

### If you've built Reth from source, provide the full command you used

RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mattsse** [comment]
I think this is a side-effect of a txpool fix, need to check what really should be emitted via the pending tx stream and how to enforce the nonce in trace call.

but you can also remove the nonce from the request.

This likely happens because these txs are emitted when being marked as pending (if there's no nonce gap) but there can still be ancestor txs (nonce - 1 > state nonce) in the pool, this would cause the nonce to high error

**Comment by mattsse** [comment]
can you also try fetching the tx count for `Pending`?

**Comment by 0xDmtri** [comment]
> I think this is a side-effect of a txpool fix, need to check what really should be emitted via the pending tx stream and how to enforce the nonce in trace call.
> 
> but you can also remove the nonce from the request.
> 
> This likely happens because these txs are emitted when being marked as pending (if there's no nonce gap) but there can still be ancestor txs (nonce - 1 > state nonce) in the pool, this would cause the nonce to high error

I tried using `None` for nonce and it was the same.

**Comment by 0xDmtri** [comment]
> can you also try fetching the tx count for `Pending`?

The tx count (`get_transaction_count()`) is the "Current nonce" on the screenshot and "Nonce from the tx" is the nonce taken from the pending tx directly. Or I am misunderstanding something?

**Comment by mattsse** [comment]
> Current nonce 

yes this defaults to `Latest`, hence I was asking you to try `Pending` instead for the block number argument

**Comment by 0xDmtri** [comment]
> > Current nonce
> 
> yes this defaults to `Latest`, hence I was asking you to try `Pending` instead for the block number argument

Oh sorry, my bad! Will do in a sec and report back, thx!

**Comment by 0xDmtri** [comment]
> > Current nonce
> 
> yes this defaults to `Latest`, hence I was asking you to try `Pending` instead for the block number argument

Here it is Ser:

<img width="780" alt="Screenshot 2023-09-28 at 21 04 43" src="https://github.com/paradigmxyz/reth/assets/33504992/84135e1e-b255-455d-b54c-4011bb7b7b33">

**Comment by 0xDmtri** [comment]
> > Current nonce
> 
> yes this defaults to `Latest`, hence I was asking you to try `Pending` instead for the block number argument

Actually what I can also do is fork at the latest commit, undo the txpool fix and test if it indeed caused this

**Comment by 0xDmtri** [comment]
@mattsse 
Just checked and can confirm that it is not caused by #4599.
Will keep digging.

**Comment by 0xDmtri** [comment]
@mattsse 
This commit caused it: #4518

**Comment by mattsse** [comment]
hmm, not obvious to me how this change would result in the error mentioned above

**Comment by 0xDmtri** [comment]
> hmm, not obvious to me how this change would result in the error mentioned above

yh, completely agree. Just a cleaner way to get current opcode and PC... this is killing me =(

**Comment by 0xDmtri** [comment]
The issue fixed itself after 88cf81fe commit.
